File: app/src/main/java/com/maldEnz/ps/presentation/gamecontroller/FireModel.java
package com.maldEnz.ps.presentation.gamecontroller;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;

import com.maldEnz.ps.R;

import java.util.Random;

public class FireModel {
    Bitmap[] fires = new Bitmap[3];
    int fireFrame = 0;
    int fireX, fireY, fireSpeed;
    Random random;

    public FireModel(Context context){
        fires[0] = BitmapFactory.decodeResource(context.getResources(), R.drawable.fire_0);
        fires[1] = BitmapFactory.decodeResource(context.getResources(), R.drawable.fire_1);
        fires[2] = BitmapFactory.decodeResource(context.getResources(), R.drawable.fire_2);
        random = new Random();
        resetPosition();
    }

    public void resetPosition(){
        fireX = random.nextInt(GameView.dWidth - getFireWidth());
        fireY = -200 + random.nextInt(600) * -1;
        fireSpeed = 35 + random.nextInt(16);
    }

    public Bitmap getFire(int fireFrame){
        return fires[fireFrame];
    }

    public int getFireWidth(){
        return fires[0].getWidth();
    }

    public int getFireHeight(){
        return fires[0].getHeight();
    }
}

File: app/src/main/java/com/maldEnz/ps/presentation/gamecontroller/FireModel.kt
package com.maldEnz.ps.presentation.gamecontroller

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import com.maldEnz.ps.R
import java.util.Random

class FireModel(context: Context) {
    val fire = Array<Bitmap>(3) {index ->
        BitmapFactory.decodeResource(context.resources, R.drawable.fire_0 + index)
        BitmapFactory.decodeResource(context.resources, R.drawable.fire_1 + index)
        BitmapFactory.decodeResource(context.resources, R.drawable.fire_2 + index)
    }
    var fireFrame = 0
    var fireX: Int? = null
    var fireY: Int? = null
    var fireSpeed: Int? = null
    var random = Random()

    init {
        resetPosition()
    }


    fun resetPosition(){
       // fireX = random.nextInt(GameView.dWidth - getFireWidth())
        fireY = -200 + random.nextInt(600) * -1
        fireSpeed = 35 + random.nextInt(16)
    }

    fun getFire(fireFrame: Int): Bitmap? {
        return fire.getOrNull(fireFrame)
    }

    fun getFireWidth() : Int{
        return fire[0]!!.width
    }

    fun getFireHeight(): Int{
        return fire[0]!!.height
    }

}